barrels/LogMonkey/README.md

```diff
@@ -0,0 +1,77 @@
+```
+using LogMonkey as Log;
+```
+Within you're project's code you can log messages, variables and Exceptions via the available APIs.
+```
+using LogMonkey as Log;
+
+class MyClassOfThings {
+
+    function aFunctionOfLogic(aParameter) {
+        Log.Debug.logMessage("MyClassOfThings", "aFunctionOfLogic()");
+        Log.Debug.logVariable("aFunctionOfLogic() param", "aParameter", aParameter);
+        ...
+        try {
+            ...do something horribly, horribly wrong...
+        } catch (ex) {
+            Log.Error.logException("Attempting to do XYZ", ex);
+        }
+        ...
+        if (aLocalVariable == aPotentiallyProblematicValue) {
+            Log.Warn.logVariable("aFunctionOfLogic()", "aLocalVariable", aLocalVariable);
+        }
+        ...
+    }
+
+}
+```
+
+## Viewing and Parsing Logs
+
+LogMonkey makes use of `Toybox.System.println()` calls to log information. This means that if your app also makes use of `pinrtln()` calls they will show up alongside the log entries.
+
+### Output Format
+
+LogMonkey logs each line in the file in the following format: `(log format version)[timestamp] {log level} tag: message`. Details for each block of this format are given in the table below.
+| Block              | Format                                                 | Description                                                                         |
+| ------------------ | ------------------------------------------------------ | ----------------------------------------------------------------------------------- |
+| log format version | A value beginning with "lfm" then a number             | The output format version of the line in the log.                                   |
+| timestamp          | MM-DD-YYYY HH:MM:SS                                    | A time stamp of when the log entry occurred.                                        |
+| log level          | Regular Expression: `[a-zA-Z0-9_]`                     | The priority level of the log entry. LogMonkey supports values of `D`, `W` and `E`. |
+| tag                | A string containing any character except a colon (`:`) | A brief tag value that can be used to quickly filter and find specific log entries. |
+| message            | Any string value                                       | The detailed message being recorded in the log entry.                               |
+
+### Viewing Logs in Eclipse
+
+When you run an app via Eclipse the run console will show the output of log calls. You will see the information your app is logging in the raw LogMonkey output format.
+![Image of run console](run_console.png)
+
+### Viewing Logs on a Device
+
+When running your app on a physical device the log statements get added to the app's log file. The app's log file can be viewed by plugging your device into your computer and navigating to `<device root>/GARMIN/APPS/LOGS`. The app log file will match the name of the app prg file (located in `<device root>/GARMIN/APPS`.
```

Comment:
Should this mention that that the user will need to create the log file in order for the logging to appear there?

Reply:
Probably. I'll add that.

Reply:
I'm not seeing anything additional here.

Reply:
I think I missed this when rebasing my change. The last commit I pushed adds this bit back in.